src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/ReadOperation.java

```diff
@@ -53,6 +53,11 @@ public interface ReadOperation extends Operation {
           "query_null_values['%s']",
           "query_timestamp_values['%s']");
   int SORT_INDEX_COLUMNS_SIZE = sortIndexColumns.size();
+
+  double TOTAL_TOKEN_RANGE = Math.pow(2, 64);
+  double MAX_TOKEN = Math.pow(2, 63) - 1;
```

Comment:
Does it make sense to have these as `double`s instead of as `long`s?

long MAX_TOKEN = Long.MAX_VALUE;
long MIN_TOKEN = Long.MIN_VALUE;

(I think)

considering they are compared to `long`/`Long` values later on.

Reply:
We went back and forth on this but as @Yuqi-Du identified, we actually need to perform calculations that can potentially exceed Long.MAX_VALUE so it's safer to treat as doubles.

Reply:
Ok. Just be aware of `double` NOT being able to accurately represent all `long` values (only up to 2^53). This is why Javascript has all the "fun" issues wrt serializing of 64-bit long values.

I doubt this will be problematic either way, and implementation can be changed if we do see problems.